torchao/dtypes/uintx/plain_layout.py

```diff
@@ -266,3 +266,58 @@ def _linear_int8_act_int8_weight_impl(input_tensor, weight_tensor, bias):
     if bias is not None:
         y += bias
     return y
+
+
+def _linear_asym_int8_act_sym_int8_weight_check(input_tensor, weight_tensor, bias):
+    return (
+        isinstance(input_tensor, AffineQuantizedTensor)
+        and _aqt_is_int8(input_tensor)
+        # ZeroPointDomain.NONE did not work for weight in int8_dynamic_activation_int8_weight
+        # Uncommenting the next line works in eager mode, but Dynamo runs into a problem with it.
+        #and torch.equal(weight_tensor.tensor_impl.zero_point, torch.zeros_like(weight_tensor.tensor_impl.zero_point))
```

Comment:
This line is a blocker. Uncommenting it results in a Dynamo related error probably because the data of the weight tensor can't be accessed during Dynamo tracing. `ZeroPointDomain.NONE` did not work for weight in `int8_dynamic_activation_int8_weight`.

Keeping this line commented is incorrect, since this approach is only meant for symmetric activation of weights. So, need to find another way to ascertain here that the weights are symmetrically quantized.

Reply:
is this because of the specific way you are comparing the tensors? 

also what is the error for `ZeroPointDomain.NONE`?

Reply:
Sorry my late response, @jerryzh168!
Thanks for taking a look!


> is this because of the specific way you are comparing the tensors?

I'm using `torch.equal` to compare tensors. This error is reported in logs generated with `TORCH_LOGS="+dynamo"` -

```
Traceback (most recent call last):
  File "torch/utils/_stats.py", line 21, in wrapper
    return fn(*args, **kwargs)
  File "torch/_subclasses/fake_tensor.py", line 1276, in __torch_dispatch__
    return self.dispatch(func, types, args, kwargs)
  File "torch/_subclasses/fake_tensor.py", line 1817, in dispatch
    return self._cached_dispatch_impl(func, types, args, kwargs)
  File "torch/_subclasses/fake_tensor.py", line 1387, in _cached_dispatch_impl
    output = self._dispatch_impl(func, types, args, kwargs)
  File "torch/_subclasses/fake_tensor.py", line 2355, in _dispatch_impl
    op_impl_out = op_impl(self, func, *args, **kwargs)
  File "torch/_subclasses/fake_impls.py", line 546, in data_dep
    raise DataDependentOutputException(func)
torch._subclasses.fake_tensor.DataDependentOutputException: aten.equal.default
```

This runtime error is reported when the process exits -

```
torch._dynamo.exc.InternalTorchDynamoError: RuntimeError: Cannot access data pointer of Tensor (e.g. FakeTensor, FunctionalTensor). If you're using torch.compile/export/fx, it is likely that we are erroneously tracing into a custom kernel. To fix this, please wrap the custom kernel into an opaque custom op. Please see the following for details: https://pytorch.org/tutorials/advanced/custom_ops_landing_page.html
```

Reply:
> also what is the error for `ZeroPointDomain.NONE`?

Runtime error at https://github.com/pytorch/ao/blob/039cef4ad546716aa04cd54c461feb173f7fe403/torchao/quantization/quant_primitives.py#L895

Reply:
oh, this check seems wrong: https://github.com/pytorch/ao/blob/039cef4ad546716aa04cd54c461feb173f7fe403/torchao/quantization/quant_primitives.py#L891 it should be zero_point_domain != ZeroPointDomain.NONE

Reply:
Thanks again for your help, @jerryzh168!

It looks like conditions pertaining to both `None` and `ZeroPointDomain.NONE` are required here, as `None` is being exclusively used for FP8.

I changed the implementation to return `None` for `zero_point` if `zero_point_domain` is `ZeroPointDomain.NONE`.

It seems most code-coverage is tested through CUDA UTs, so if CUDA UTs would fail, I'll revise the implementation to return a zero point even when `zero_point_domain` would be `ZeroPointDomain.NONE`.